src/program.py

```diff
@@ -200,106 +210,117 @@ def serialize(self) -> str:
-    def __init__(self, lid: int, sort: Instruction, op1: Instruction, op2: Instruction):
+    def __init__(self, lid: int, sort: Sort, op1: Instruction, op2: Instruction):
         super().__init__(lid, "eq", [sort, op1, op2])
 
 class Neq(Instruction):
-    def __init__(self, lid: int, sort: Instruction, op1: Instruction, op2: Instruction):
+    def __init__(self, lid: int, sort: Sort, op1: Instruction, op2: Instruction):
         super().__init__(lid, "neq", [sort, op1, op2])   
 
 class Ugt(Instruction):
-    def __init__(self, lid: int, sort: Instruction, op1: Instruction, op2: Instruction):
+    def __init__(self, lid: int, sort: Sort, op1: Instruction, op2: Instruction):
         super().__init__(lid, "ugt", [sort, op1, op2])
 
 class Sgt(Instruction):
-    def __init__(self, lid: int, sort: Instruction, op1: Instruction, op2: Instruction):
+    def __init__(self, lid: int, sort: Sort, op1: Instruction, op2: Instruction):
         super().__init__(lid, "sgt", [sort, op1, op2])
     
 class Ugte(Instruction):
-    def __init__(self, lid: int, sort: Instruction, op1: Instruction, op2: Instruction):
+    def __init__(self, lid: int, sort: Sort, op1: Instruction, op2: Instruction):
         super().__init__(lid, "ugte", [sort, op1, op2])
 
 class Sgte(Instruction):
-    def __init__(self, lid: int, sort: Instruction, op1: Instruction, op2: Instruction):
+    def __init__(self, lid: int, sort: Sort, op1: Instruction, op2: Instruction):
         super().__init__(lid, "sgte", [sort, op1, op2])
 
 class Ult(Instruction):
-    def __init__(self, lid: int, sort: Instruction, op1: Instruction, op2: Instruction):
+    def __init__(self, lid: int, sort: Sort, op1: Instruction, op2: Instruction):
         super().__init__(lid, "ult", [sort, op1, op2])
 
 class Slt(Instruction):
-    def __init__(self, lid: int, sort: Instruction, op1: Instruction, op2: Instruction):
+    def __init__(self, lid: int, sort: Sort, op1: Instruction, op2: Instruction):
         super().__init__(lid, "slt", [sort, op1, op2])
 
 class Ulte(Instruction):
-    def __init__(self, lid: int, sort: Instruction, op1: Instruction, op2: Instruction):
+    def __init__(self, lid: int, sort: Sort, op1: Instruction, op2: Instruction):
         super().__init__(lid, "ulte", [sort, op1, op2])
 
 class Slte(Instruction):
-    def __init__(self, lid: int, sort: Instruction, op1: Instruction, op2: Instruction):
+    def __init__(self, lid: int, sort: Sort, op1: Instruction, op2: Instruction):
         super().__init__(lid, "slte", [sort, op1, op2])
+
+class Uext(Instruction):
+    def __init__(self, lid: int, sort: Sort, op: Instruction, width: int, name: str):
+        super().__init__(lid, "uext", [sort, op, width, name])
+        # Handle btor uexts which are creating name aliases
+        self.width: int = width
+        self.renaming = False
+        if self.width == 0:
+            self.renaming = True
+            self.name = name
+            self.aliasid = op.lid
```

Comment:
new line

Also lol I didn't know btro2 supported aliasing like this, I just forced myself to get rid of everything before emission in CIRCT. Nice

Reply:
This is really just something that yosys does when it wants to say that one wire (e.g., module port) is connected with another. 

Now that I check, it does seem to be a part of the btor2 spec though: https://fmv.jku.at/papers/NiemetzPreinerWolfBiere-CAV18.pdf